eth/tracers/native/call.go

```diff
@@ -166,6 +166,11 @@ func (t *callTracer) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64, sco
 		stack := scope.Stack
 		stackData := stack.Data()
 
+		stackRequired := (vm.LOG4 - vm.LOG0) + 2
+		if len(stackData) < int(stackRequired) {
```

Comment:
Is this needed? AFAIK `CaptureState` is called after the stack validation

Reply:
it's called before stack validation: https://github.com/ethereum/go-ethereum/blob/master/core/vm/interpreter.go#L159

Reply:
That case is only when op execution failed. These 2 are the main invocations, if memory expansion is expected:

https://github.com/ethereum/go-ethereum/blob/94ff7219114540263c76d1895618041959fae1f9/core/vm/interpreter.go#L217

Otherwise:

https://github.com/ethereum/go-ethereum/blob/94ff7219114540263c76d1895618041959fae1f9/core/vm/interpreter.go#L224

Reply:
Looks like what happens in the case of stack underflow when executing `LOG`, is that the "op execution failed" case is called after the stack underflow error occurs in the interpreter loop.

So afaict, adding a stack check seems necessary?

Reply:
Ah pardon you're right. I was assuming callTracer aborts if the error passed to CaptureState is non-nil. IMO that should be the fix and we shouldn't process the logs at all if the interpreter passed in an error.

Reply:
Have applied this "return early if err != nil" logic to the native call tracer.  Should this be applied across all tracers?

Reply:
> Should this be applied across all tracers?

Prestate tracer also has this issue. Please apply there too

Reply:
> Prestate tracer also has this issue. 

@s1na I found one case, via CREATE2, where the early return fixed it. Was there any other case?